### PR TITLE
Jetpack Cloud: Add golden ticket dialog

### PR DESCRIPTION
**Pull request by robertf4**
## Proposed Changes

This PR adds a dialog to let a user accept a golden ticket.

## Testing Instructions

In Jetpack cloud, visit `/golden-token`. Check the styling, copy, animations, etc. The business logic will be added in a later PR.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	const redeemToken = () => {
+		if ( videoRef.current ) {
+			videoRef.current.play();
+		}
+		setIsAnimating( true );
+	};
+
+	return (
+		<div className={ classnames( 'golden-token-dialog', { animating: isAnimating } ) }>
+			<svg
```
Comment (rcoll) [inline client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx:26]: For cleanliness, would it make sense to make this SVG its own component?

File: client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+					<p>
+						{ translate(
+							'Your Jetpack Gold Token provides a lifetime license for this website and includes the following products:'
+						) }
+					</p>
+				</div>
+
+				<div className="golden-token-dialog__jetpack-products">
+					<div>
+						<svg
```
Comment (rcoll) [inline client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx:74]: Same as the others, would it make sense to make this SVG its own component?

File: client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+						<h3>{ translate( 'VaultPress Backup' ) }</h3>
+						<p>
+							{ translate(
+								'Save every change and get back online quickly with one‑click restores.'
+							) }
+						</p>
+					</div>
+					<div>
+						<svg
```
Comment (rcoll) [inline client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx:113]: Same as the others, would it make sense to make this SVG its own component?

File: config/jetpack-cloud-stage.json [hunk 4]
```diff
@@ -78,6 +78,7 @@
 		"jetpack-cloud-plugin-management": true,
 		"jetpack-cloud-pricing": true,
 		"jetpack-cloud-settings": true,
+		"jetpack-cloud-golden-token": false,
```
Comment (rcoll) [inline config/jetpack-cloud-stage.json]: I'd actually make this true so that it can be shown off in staging immediately.

**Review by rcoll — APPROVED** [review]
I left a couple nitpicks but this looks fine to ship and iterate on! Nice work!

File: client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx (hunk 1 above)
Comment (robertf4) [reply client/jetpack-cloud/sections/golden-token/golden-token-dialog/index.tsx:26]: Yes, for this and all the others I plan to go back through and switch them to existing icon components.

File: config/jetpack-cloud-stage.json (hunk 4 above)
Comment (robertf4) [reply config/jetpack-cloud-stage.json]: Will do!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7897584

Hi @robertf4, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Hey, %(userName)s`
- `Your exclusive Jetpack Experience awaits`
- `You have been gifted a Jetpack Golden Token. This unlocks a lifetime of Jetpack powers for your website.`
- `Redeem your token`
- `Super powers are ready!`
- `Your Jetpack Golden Token provides a lifetime license for this website and includes the following products:`
- `Explore your new powers`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7897584).

**Pull request merged**